src/ray/raylet/node_manager.cc

```diff
@@ -1595,7 +1595,7 @@ void NodeManager::HandleReturnWorker(const rpc::ReturnWorkerRequest &request,
                                      rpc::SendReplyCallback send_reply_callback) {
   // Read the resource spec submitted by the client.
   auto worker_id = WorkerID::FromBinary(request.worker_id());
-  std::shared_ptr<Worker> worker = std::move(leased_workers_[worker_id]);
+  std::shared_ptr<Worker> worker = leased_workers_[worker_id];
```

Comment:
Somewhat conflicted about this one since we do delete it in both cases below. Maybe we could just keep the `std::move` and do the delete just below it? Would be nice if there was a "delete-and-move" interface.

Reply:
I'm not sure what the correct thing here should be (at least it's not obvious at first glance, I haven't spent too much time looking at it) so happy to remove it if that might be safer.

Reply:
It's not immediately obvious to me either. Let's leave your change as-is.

Reply:
Ok